Rename LazyLineBreakIterator to CachedTextBreakIteratorFactory

litherum opened the pull request:
#### 522c36c64af3fa1b643f3f5c7458571b96b5af9b
<pre>
Rename LazyLineBreakIterator to CachedTextBreakIteratorFactory
<a href="https://bugs.webkit.org/show_bug.cgi?id=257696">https://bugs.webkit.org/show_bug.cgi?id=257696</a>
rdar://110237360

Reviewed by Cameron McCormack.

LazyLineBreakIterator isn&apos;t actually an iterator. It&apos;s really just a class that holds all the parameters
necessary to create an iterator. We use it so that we can populate these parameters at the top of layout
code, but if it turns out we don&apos;t actually need an iterator (e.g. the element is empty or something)
we don&apos;t actually create the iterator until we need to. Therefore, it really should be called a factory,
rather than an iterator itself.

* Source/WTF/wtf/text/TextBreakIterator.h:
(WTF::CachedTextBreakIteratorFactory::CachedTextBreakIteratorFactory):
(WTF::LazyLineBreakIterator::PriorContext::PriorContext): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::lastCharacter const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::secondToLastCharacter const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::set): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::update): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::reset): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::length const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::characters const): Deleted.
(WTF::LazyLineBreakIterator::LazyLineBreakIterator): Deleted.
(WTF::LazyLineBreakIterator::stringView const): Deleted.
(WTF::LazyLineBreakIterator::mode const): Deleted.
(WTF::LazyLineBreakIterator::get): Deleted.
(WTF::LazyLineBreakIterator::resetStringAndReleaseIterator): Deleted.
(WTF::LazyLineBreakIterator::priorContext const): Deleted.
(WTF::LazyLineBreakIterator::priorContext): Deleted.
* Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp:
(WebCore::Layout::moveToNextBreakablePosition):
(WebCore::Layout::InlineItemsBuilder::handleTextContent):
* Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp:
(WebCore::Layout::endsWithSoftWrapOpportunity):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::findNextBreakablePosition):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.h:
* Source/WebCore/rendering/BreakLines.h:
(WebCore::nextBreakablePosition):
(WebCore::nextBreakablePositionKeepingAllWords):
(WebCore::nextBreakablePositionKeepingAllWordsIgnoringNBSP):
(WebCore::nextBreakablePositionIgnoringNBSP):
(WebCore::nextBreakablePositionWithoutShortcut):
(WebCore::nextBreakablePositionIgnoringNBSPWithoutShortcut):
(WebCore::nextBreakablePositionBreakCharacter):
(WebCore::isBreakable):
* Source/WebCore/rendering/RenderBlockFlow.h:
(WebCore::RenderBlockFlow::cachePriorCharactersIfNeeded):
* Source/WebCore/rendering/RenderRubyBase.cpp:
(WebCore::RenderRubyBase::cachePriorCharactersIfNeeded):
* Source/WebCore/rendering/RenderRubyBase.h:
* Source/WebCore/rendering/RenderRubyRun.cpp:
(WebCore::RenderRubyRun::updatePriorContextFromCachedBreakIterator const):
(WebCore::RenderRubyRun::canBreakBefore const):
* Source/WebCore/rendering/RenderRubyRun.h:
* Source/WebCore/rendering/RenderRubyText.cpp:
(WebCore::RenderRubyText::canBreakBefore const):
* Source/WebCore/rendering/RenderRubyText.h:
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::computePreferredLogicalWidths):
* Source/WebCore/rendering/line/LineBreaker.h:
* Source/WebKitLegacy/ios/Misc/WebUIKitSupport.mm:
(WebKitGetLastLineBreakInBuffer):
* Tools/TestWebKitAPI/Tests/WTF/TextBreakIterator.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/264858@main">https://commits.webkit.org/264858@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39fc340dcbee341bfcf0658212d78589e55ea9a3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8904 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9192 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9410 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10557 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8884 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8912 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11179 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9159 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11736 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9050 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10040 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10716 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7336 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8142 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15632 "Passed tests") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/7659 "Found 1 jsc stress test failure: wasm.yaml/wasm/stress/simple-inline-stacktrace-with-catch.js.wasm-eager") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8445 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8290 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11620 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/8547 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8781 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7161 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/9071 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8037 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2159 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2153 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12248 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/9306 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8529 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2278 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

heycam commented on Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp:
```diff
@@ -63,7 +63,7 @@ static std::optional<WhitespaceContent> moveToNextNonWhitespacePosition(StringVi
 }
 
-static unsigned moveToNextBreakablePosition(unsigned startPosition, LazyLineBreakIterator& lineBreakIterator, const RenderStyle& style)
+static unsigned moveToNextBreakablePosition(unsigned startPosition, CachedTextBreakIteratorFactory& lineBreakIterator, const RenderStyle& style)
```
In the same vein as the class renaming, here, and elsewhere in this patch, you might consider renaming arguments/variables from `xxxIterator` to something else.

heycam commented on Source/WebCore/rendering/RenderBlockFlow.h:
```diff
@@ -462,7 +462,7 @@ class RenderBlockFlow : public RenderBlock {
     virtual void computeColumnCountAndWidth();
 
-    virtual void cachePriorCharactersIfNeeded(const LazyLineBreakIterator&) {};
+    virtual void cachePriorCharactersIfNeeded(const CachedTextBreakIteratorFactory&) { };
```
Can drop the semicolon while touching this line.

litherum committed:
Rename LazyLineBreakIterator to CachedTextBreakIteratorFactory
https://bugs.webkit.org/show_bug.cgi?id=257696
rdar://110237360

Reviewed by Cameron McCormack.

LazyLineBreakIterator isn't actually an iterator. It's really just a class that holds all the parameters
necessary to create an iterator. We use it so that we can populate these parameters at the top of layout
code, but if it turns out we don't actually need an iterator (e.g. the element is empty or something)
we don't actually create the iterator until we need to. Therefore, it really should be called a factory,
rather than an iterator itself.

* Source/WTF/wtf/text/TextBreakIterator.h:
(WTF::CachedTextBreakIteratorFactory::CachedTextBreakIteratorFactory):
(WTF::LazyLineBreakIterator::PriorContext::PriorContext): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::lastCharacter const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::secondToLastCharacter const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::set): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::update): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::reset): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::length const): Deleted.
(WTF::LazyLineBreakIterator::PriorContext::characters const): Deleted.
(WTF::LazyLineBreakIterator::LazyLineBreakIterator): Deleted.
(WTF::LazyLineBreakIterator::stringView const): Deleted.
(WTF::LazyLineBreakIterator::mode const): Deleted.
(WTF::LazyLineBreakIterator::get): Deleted.
(WTF::LazyLineBreakIterator::resetStringAndReleaseIterator): Deleted.
(WTF::LazyLineBreakIterator::priorContext const): Deleted.
(WTF::LazyLineBreakIterator::priorContext): Deleted.
* Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp:
(WebCore::Layout::moveToNextBreakablePosition):
(WebCore::Layout::InlineItemsBuilder::handleTextContent):
* Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp:
(WebCore::Layout::endsWithSoftWrapOpportunity):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::findNextBreakablePosition):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.h:
* Source/WebCore/rendering/BreakLines.h:
(WebCore::nextBreakablePosition):
(WebCore::nextBreakablePositionKeepingAllWords):
(WebCore::nextBreakablePositionKeepingAllWordsIgnoringNBSP):
(WebCore::nextBreakablePositionIgnoringNBSP):
(WebCore::nextBreakablePositionWithoutShortcut):
(WebCore::nextBreakablePositionIgnoringNBSPWithoutShortcut):
(WebCore::nextBreakablePositionBreakCharacter):
(WebCore::isBreakable):
* Source/WebCore/rendering/RenderBlockFlow.h:
(WebCore::RenderBlockFlow::cachePriorCharactersIfNeeded):
* Source/WebCore/rendering/RenderRubyBase.cpp:
(WebCore::RenderRubyBase::cachePriorCharactersIfNeeded):
* Source/WebCore/rendering/RenderRubyBase.h:
* Source/WebCore/rendering/RenderRubyRun.cpp:
(WebCore::RenderRubyRun::updatePriorContextFromCachedBreakIterator const):
(WebCore::RenderRubyRun::canBreakBefore const):
* Source/WebCore/rendering/RenderRubyRun.h:
* Source/WebCore/rendering/RenderRubyText.cpp:
(WebCore::RenderRubyText::canBreakBefore const):
* Source/WebCore/rendering/RenderRubyText.h:
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::computePreferredLogicalWidths):
* Source/WebCore/rendering/line/LineBreaker.h:
* Source/WebKitLegacy/ios/Misc/WebUIKitSupport.mm:
(WebKitGetLastLineBreakInBuffer):
* Tools/TestWebKitAPI/Tests/WTF/TextBreakIterator.cpp:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/264858@main
